jigglypuffRL/deeprl/ddpg/ddpg.py

```diff
@@ -272,12 +268,12 @@ def learn(self):
                         states, actions, next_states, rewards, dones
                     )
 
-            if t >= self.start_update and t % self.save_interval == 0:
-                if self.save_name is None:
-                    self.save_name = self.network_type
-                self.save_version = int(t / self.save_interval)
-                self.checkpoint["weights"] = self.ac.state_dict()
-                self.save(self)
+            if self.save_model is not None:
+                if t >= self.start_update and t % self.save_interval == 0:
+                    self.checkpoint["weights"] = self.ac.state_dict()
+                    self.save(self, self.save_model, t)
+
+        self.run_num += 1
```

Comment:
This would break if I’m not calling learn using the same object. Could you check the checkpoint directory instead?

```
files = os.listdir(ckpt_dir)
ckpt_name = ... # without run-I’d
relevant = [file for file in files if ckpt_name in file]
relevant = sorted(relevant)
last_run_id = relevant[-1][-1]
self.run_id = last_run_id + 1
```

Reply:
I still have to sort a few things. It's not complete yet.

Reply:
I made a separate dict called log which I'm pickling each time. It'll be saved in whatever the dorecrory is so it works fine.